src/libs/actions/PersonalDetails.js

```diff
@@ -243,7 +243,7 @@ function fetchCurrencyPreferences() {
         isRetrievingCurrency: true,
     });
 
-    API.GetPreferredCurrency({...coords})
+    API.GetPreferredCurrency({...coords, doNotRetry: true})
```

Comment:
Yes, as I mentioned in [this comment](https://github.com/Expensify/Expensify.cash/pull/4019#discussion_r670600927), `isRetrievingCurrency` is never set back to false because `API.GetPreferredCurrency` never resolves.